pyproject.toml

```diff
@@ -0,0 +1,115 @@
+[build-system]
+build-backend = "hatchling.build"
+requires = [
+    "hatch-vcs",
+    "hatchling",
+]
+
+[project]
+dependencies = [
+    "array_api_strict",
+    "jax",
+    "numpy",
+    "pytest",
+    "pytest-benchmark",
+]
+description = "Benchmarks for glass-dev/glass"
+dynamic = [
+    "version",
+]
+maintainers = [
+    {email = "n.tessore@ucl.ac.uk", name = "Nicolas Tessore"},
+]
+name = "glass-benchmarks"
+readme = "README.md"
+requires-python = ">=3.10"
+license.text = "MIT License"
+
+[tool.hatch]
+build.hooks.vcs.version-file = "tests/_version.py"
+build.targets.sdist.exclude = [
+    ".*",
+    "noxfile.py",
+    "tests/*",
+]
+version.source = "vcs"
+
+[tool.mypy]
+disallow_untyped_decorators = false
+enable_error_code = [
+    "ignore-without-code",
+    "redundant-expr",
+    "truthy-bool",
+]
+explicit_package_bases = true
+strict = true
+warn_unreachable = true
+
+[tool.pytest.ini_options]
+addopts = [
+    "--strict-config",
+    "--strict-markers",
+    "-ra",
+    "-v",
+]
+filterwarnings = [
+    "ignore::DeprecationWarning",
+]
+log_cli_level = "DEBUG"
+minversion = "6.0"
+testpaths = [
+    "tests",
+]
+xfail_strict = true
+
+[tool.ruff]
+fix = true
+force-exclude = true
+show-fixes = true
+lint.future-annotations = true
+lint.ignore = [
+    "COM812", # missing-trailing-comma (ruff-format recommended)
+    "D203", # one-blank-line-before-class
+    "D212", # blank-line-before-class
+    "ISC001", # single-line-implicit-string-concatenation (ruff-format recommended)
+]
+lint.isort = {known-first-party = [
+    "glass",
+], known-third-party = [
+    "jax",
+], section-order = [
+    "future",
+    "standard-library",
+    "third-party",
+    "array-api",
+    "cosmo",
+    "first-party",
+    "local-folder",
+], sections = {"array-api" = [
+    "array_api_compat",
+    "array_api_extra",
+    "array_api_strict",
+], "cosmo" = [
+    "camb",
+    "cosmology",
+]}}
+lint.per-file-ignores = {"tests*" = [
+    "D100", # undocumented-public-module
+    "INP001", # implicit-namespace-package
+    "PLR2004", # magic-value-comparison
+    "S101", # assert
+]}
```

Comment:
Best to add to these as we go rather than copying. Best to have as many enabled as possible IMO.